edit-post/components/layout/index.js

```diff
@@ -84,6 +86,8 @@ function Layout( {
 					onClose={ closePublishSidebar }
 					forceIsDirty={ hasActiveMetaboxes }
 					forceIsSaving={ isSaving }
+					renderPrePublishExtension={ () => <PluginPrePublishPanel.Slot /> }
```

Comment:
This will cause render reconciliation more than necessary because it's creating a new function each render.

Why not hoist to a top-level variable? Or could `renderPrePublishExtension` be interpreted as a component, so we could pass the `PluginPrePublishPanel.Slot` reference directly instead of a function?

Reply:
Interesting, I would also think we would cause render reconciliation more than necessary because we are passing new props on each rerender, but the react samples for the  Render Props pattern use something similar https://reactjs.org/docs/render-props.html, they are also creating inline functions and passing them as props. I see this practice being used widely. I'm curious if react has some an optimization for the inline functions rendering React components.

Reply:
We probably can pass those slots as components in this case. Opening PR soon.

Reply:
Done: #6938.